src/cascadia/TerminalSettingsModel/ProfileEntry.h

```diff
@@ -0,0 +1,54 @@
+/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- FolderEntry.h
+
+Abstract:
+- A profile entry in the "new tab" dropdown menu, referring to
+    a single profile.
+
+Author(s):
+- Floris Westerman - August 2022
+
+--*/
+#pragma once
+
+#include "pch.h"
+#include "NewTabMenuEntry.h"
+#include "ProfileEntry.g.h"
+
+#include "Profile.h"
+
+namespace winrt::Microsoft::Terminal::Settings::Model::implementation
+{
+    struct ProfileEntry : ProfileEntryT<ProfileEntry, NewTabMenuEntry>
+    {
+    public:
+        ProfileEntry() noexcept;
+        explicit ProfileEntry(const winrt::hstring& profile) noexcept;
+
+        Json::Value ToJson() const override;
+        static com_ptr<NewTabMenuEntry> FromJson(const Json::Value& json);
+
+        // In JSON, only a profile name (guid or string) can be set;
+        // but the consumers of this class would like to have direct access
+        // to the appropriate Model::Profile. Therefore, we have a read-only
+        // property ProfileName that corresponds to the JSON value, and
+        // then CascadiaSettings::_resolveNewTabMenuProfiles() will populate
+        // the Profile and ProfileIndex properties appropriately
+        winrt::hstring ProfileName() const noexcept { return _ProfileName; };

```

Comment:
~Do we need a separate member for `_ProfileName`, or~

Okay I think I get it. `_ProfileName` is just "name or guid from the raw json"